File: web/src/app/modules/pos/services/pos-sale.service.ts
// web/src/app/modules/pos/services/pos-sale.service.ts

import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from '../../../core/services/api.service';

@Injectable({
  providedIn: 'root'
})
export class PosSaleService {
  constructor(private apiService: ApiService) {}

  registrarVenta(idSucursal: number, metodoPago: string, items: { id_variante: number; cantidad: number }[]): Observable<any> {
    return this.apiService.post('/sales/ventas-pos/registrar/', {
      id_sucursal: idSucursal,
      metodo_pago: metodoPago,
      items,
    });
  }
}

File: mobile/lib/features/pos/data/pos_sale_service.dart
// mobile/lib/features/pos/data/pos_sale_service.dart

import '../../../core/network/api_client.dart';

class PosSaleService {
  final ApiClient apiClient;

  PosSaleService({required this.apiClient});

  Future<Map<String, dynamic>> registrarVenta({
    required int idSucursal,
    required String metodoPago,
    required List<Map<String, dynamic>> items,
  }) async {
    final response = await apiClient.dio.post(
      '/sales/ventas-pos/registrar/',
      data: {
        'id_sucursal': idSucursal,
        'metodo_pago': metodoPago,
        'items': items,
      },
    );
    return response.data as Map<String, dynamic>;
  }
}